graylog2-server/src/main/java/org/graylog/plugins/views/search/searchtypes/events/EventList.java

```diff
@@ -119,6 +119,11 @@ public SearchType applyExecutionContext(SearchTypeExecutionState state) {
             state.page().ifPresent(builder::page);
             state.perPage().ifPresent(builder::perPage);
             return builder.build();
+        } else if (state.limit().isPresent() && (state.offset().isEmpty()) || state.offset().get().equals(0)) {
```

Comment:
is this expression correct? Or did you want 
`if (state.limit().isPresent() && (state.offset().isEmpty() || state.offset().get().equals(0)))`?

Reply:
You did not expect the `else` clause?
My thought was that limit will be used only if page version is not specified. If page or per_page is present, this is the version that is used.

Reply:
Thanks a lot! :)